src/main.rs

```diff
@@ -148,7 +148,9 @@ fn main() -> Result<()> {
         Codec::Json => {
             let json = serde_json::from_slice::<serde_json::Value>(&buffer)
                 .map_err(|e| anyhow!("Invalid input JSON: {}", e))?;
-            (Some(json), buffer)
+            let minified_buffer =
+                serde_json::to_vec(&json).map_err(|e| anyhow!("Couldn't serialize JSON: {}", e))?;
+            (Some(json), minified_buffer)
```

Comment:
1. I *believe* that since this isn't a mutable reference that's passed in, the JSON value wouldn't be changed. My Rust knowledge isn't very deep yet so I might be wrong on that.
2. The JSON value is the in-memory representation of the content. The string representation *shouldn't* have an effect on that.
3. The JSON value is only used for dynamic scaling analysis, so minification wouldn't matter anyway 😛